%YAML 1.0
sensors:
  imu:
    topic: /sensors/bno08x/raw
    sample_rate: 300
    bus: 1

  distance:
    topic: /sensors/vl53l0x/dist
    sample_rate: 300
    bus: 1

  color:
    topic: /sensors/tcs34725/color
    sample_rate: 100
    pins:
      s2: 19
      s3: 26
      signal: 13

  camera:
    topic: /sensors/imx/image_raw
    sample_rate: 30 # is also framerate
    # input_stream: csi://0
    input_stream: "0"
    imu_sync: 1
    resolution:
      width: 640
      height: 480

  battery:
    sample_rate: 1
    topic: 
      # sensors battery
      - /sensors/ina219_0/state 
      # motors battery
      - /sensors/ina219_1/state 
    type: 
      - LIHV
      - LIPO
    cells: 
      - 3
      - 2
    capacity: 
      - 2300 # mAh
      - 1800 # mAh
    bus:
      - 1
      - 0

  bluetooth_gamepad:
    topic: "/sensors/bluetooth/buttons"
    sample_rate: 100
    device_name: "Q36 for Android"

actuators:
  motors:
    topic: /cmd_vel
    brake_topic: /brake
    use_brake: 1
    esc_channel: 0
    servo_channel: 1
    max_speed: 30
    angle_step: 5 # degrees
    speed_step: 0.05 

  flare: # led tape
    topic: /flare
    pin: 18
    # pin: 12

  led: # onboard signal leds
    topics: 
      - /led_1
      - /led_2
    pins:
      red: # linux gpio notation
        - 16
        - 17
      green:
        - 20
        - 27 
      blue:
        - 21
        - -1 # should be 22, but for mysterious reasons it doesnt work

cone_detector:
  topic: "/cone_detector/detections"
  intrinsics: "/home/user/ws/src/config/imx219_calib_data.yaml"
  sample_rate: 10
  color_upper:
    - 105
    - 140
    - 230
  color_lower:
    - 60
    - 90
    - 170
  debug_image: 1
  debug_topic: "/cone_detector/bounding_box"

mark_detector:
  topic: "/mark_detectror/detected"
  sample_rate: 10
  color_upper:
    - 120
    - 206
    - 220
  color_lower:
    - 80
    - 190
    - 190
    # - 150
    # - 150
  debug_image: 1
  debug_topic: "/mark_detector/bounding_box"

steering:
  odometry_topic: /imu_tracker/odom
  map_topic: /steering/map
  map_file: /home/user/ws/Data/maps/straight_line.txt
  bt_start_stop: 1
  mark_color_lower:
    - 3200
    - 4000
    - 2000
  mark_color_upper:
    - 4200
    - 4200
    - 3500
  linear_speed: 13
  ignore_mark_for: 5000
  use_cone_pos_after: 100000
  track_radius: 0.001

pose_logger:
  pose_topic: "/imu_tracker/odom"
  output_path: "/home/user/ws/Data/maps"
  skip_poses: 20

visualization_node:
  topics:
    - "/steering/map"
    - "/imu_tracker/odom"
